types/react-select/src/Select.d.ts

```diff
@@ -202,16 +205,17 @@ export interface NamedProps<OptionType extends OptionTypeBase = { label: string;
   /* Select the currently focused option when the user presses tab */
   tabSelectsValue?: boolean;
   /* The value of the select; reflected by the selected option */
-  value?: ValueType<OptionType>;
+  value?: ValueType<OptionType, IsMulti>;
 
   defaultInputValue?: string;
   defaultMenuIsOpen?: boolean;
-  defaultValue?: ValueType<OptionType>;
+  defaultValue?: IsMulti extends true ? ValueType<OptionType, boolean> : ValueType<OptionType, false>;
 }
 
-export interface Props<OptionType extends OptionTypeBase = { label: string; value: string }>
-    extends NamedProps<OptionType>,
-        SelectComponentsProps {}
+export interface Props<
+  OptionType extends OptionTypeBase = { label: string; value: string },
+  IsMulti extends boolean = false
```

Comment:
This did break my code though and took a while to find. out i had to add additional generic type to select since type error was not very forth coming) 

```
<Select<ModelType, true> // add true to tell select its a multi select.
```

Reply:
```
 IsMulti extends boolean = boolean
```
would have been a better backwardscompatible default.

Reply:
I don't believe this would've worked because we want `IsMulti` to be `false` if it's not set. But I don't remember for sure if that's the case, I'd have to go back and check.

Reply:
But feel free to make a PR if that works! It would mean less breakage for future people.

Reply:
Yea I just had a look at this and you are right, this doesnt seem to work. I assume because the `isMulti` prop would be `undefined` instead of strictly `false` as there might be some confusion in typescript about this?

using `false` as the default works better in the following example

https://www.typescriptlang.org/play?#code/C4TwDgpgBAaghgGwK4QCrggHgLJQgD2AgDsATAZygCMB7GhCOYgPigF4oAKXAosy4ACcUUAPxRyQgJbEA5gG0AulABcE6XICUUAD5RiSBAl1QkZCADMZEUgG4AUPZlFBFuAGNoAZQgN3wAAVBGjByHDxCEgpqOgYmVgBveygUqAA3RBRRNXhkNAwcZgdUqClybENgKWyobGLU0ks4StystR5I-ighEXFW-MhMWnpGFlVYTIGsbCLk1JpiAGEACyZZCDVODLycyfRBme02VjSaKTt7AF9HewB6W6gAVXIZWSgAA2G44neoOEoAJLlSpSKCNNyVRzuBaSboQSQAIViowAYsR2FBwrwopQvqMMXj4pwwMFQmofH5AqSwod2IlHNd7EREcimGjOEkSgsVmsNlw0kcTqIAHTAGiPMCQQSLf4QTiaK4Ku4PZ6vD5uBDkCC-f5QIEVBBVMFNSH2aHEWHM4AoxBatEYrGdaKE9EcDVa5jE6nk3wQfxBEI05iCqBJK6OK02zUQdmc+ZLVZyPlbENpEViiVSmVa+WKoA